# Preparare i dati

In questo laboratorio impareremo a trattare dati provenienti da sorgenti diverse, cercando di ottenere informazioni integrate e complete. 
Come esempio, ci concentreremo su dati riguardanti film e serie tv dei maggiori servizi di streaming. 

In [401]:
import pandas as pd

### Carico i dati di Netflix

In [402]:
netflix_titles = pd.read_csv("netflix_titles.csv")

Diamo un'occhiata a come è organizzato il dataset

In [403]:
netflix_titles.head()

Unnamed: 0        id                                title   type  \
0           0  ts300399  Five Came Back: The Reference Films   SHOW   
1           1   tm82169                                Rocky  MOVIE   
2           2   tm17823                               Grease  MOVIE   
3           3   tm69975                             Rocky II  MOVIE   
4           4  tm127384      Monty Python and the Holy Grail  MOVIE   

   release_year age_certification  runtime                 genres  \
0          1945             TV-MA       51      ['documentation']   
1          1976                PG      119     ['drama', 'sport']   
2          1978                PG      110  ['comedy', 'romance']   
3          1979                PG      119     ['drama', 'sport']   
4          1975                PG       91  ['fantasy', 'comedy']   

  production_countries  seasons  ... tmdb_popularity  tmdb_score  \
0               ['US']      1.0  ...           0.601         NaN   
1               ['US']      NaN  ...          99.713       7.783   
2               ['US']      NaN  ...          33.160       7.406   
3               ['US']      NaN  ...          74.191       7.200   
4               ['GB']      NaN  ...          20.964       7.804   

                     director  \
0                         NaN   
1            John G. Avildsen   
2              Randal Kleiser   
3          Sylvester Stallone   
4  Terry Gilliam, Terry Jones   

                                                cast         country  \
0                                                NaN   United States   
1  Sylvester Stallone, Talia Shire, Burt Young, C...   United States   
2  John Travolta, Olivia Newton-John, Stockard Ch...   United States   
3  Sylvester Stallone, Talia Shire, Burt Young, C...   United States   
4  Graham Chapman, John Cleese, Eric Idle, Terry ...  United Kingdom   

         date_added rating  duration  \
0  February 9, 2021  TV-MA  1 Season   
1    August 1, 2019     PG   120 min   
2  November 1, 2019     PG   110 min   
3    August 1, 2019     PG   119 min   
4   January 2, 2019     PG    92 min   

                                      listed_in  \
0                                      TV Shows   
1    Action & Adventure, Classic Movies, Dramas   
2    Classic Movies, Comedies, Music & Musicals   
3                         Dramas, Sports Movies   
4  Action & Adventure, Classic Movies, Comedies   

                                         description  
0  This collection includes 12 World War II-era p...  
1  Sylvester Stallone shot to fame as Rocky Balbo...  
2  John Travolta and Olivia Newton-John star in t...  
3  Featuring a rousing climax, this engaging sequ...  
4  The Monty Python comedy clan skewers King Arth...  

[5 rows x 23 columns]

ed allo schema (notate che con head() alcune delle colonne, per problemi di spazio, non compaiono)

In [404]:
netflix_titles.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3645 entries, 0 to 3644
Data columns (total 23 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            3645 non-null   int64  
 1   id                    3645 non-null   object 
 2   title                 3645 non-null   object 
 3   type                  3645 non-null   object 
 4   release_year          3645 non-null   int64  
 5   age_certification     2003 non-null   object 
 6   runtime               3645 non-null   int64  
 7   genres                3645 non-null   object 
 8   production_countries  3645 non-null   object 
 9   seasons               1453 non-null   float64
 10  imdb_id               3417 non-null   object 
 11  imdb_score            3378 non-null   float64
 12  imdb_votes            3374 non-null   float64
 13  tmdb_popularity       3612 non-null   float64
 14  tmdb_score            3529 non-null   float64
 15  director             

### Carico i dati di Disney+

In [405]:
disney_titles = pd.read_csv("disney_titles.csv")

In [406]:
disney_titles.head()

Unnamed: 0        id                                   title show_type  \
0           0   tm61729  The Adventures of Ichabod and Mr. Toad     MOVIE   
1           1   tm61052                              Cinderella     MOVIE   
2           2  tm190870                              Cinderella     MOVIE   
3           3   tm87946                                   Dumbo     MOVIE   
4           4  tm207119                                   Dumbo     MOVIE   

   release_year age_certification  runtime  \
0          1950                 G       68   
1          1950                 G       74   
2          1997                 G       88   
3          1941                 G       64   
4          2019                PG      112   

                                              genres production_countries  \
0  ['horror', 'fantasy', 'animation', 'comedy', '...               ['US']   
1      ['fantasy', 'animation', 'family', 'romance']               ['US']   
2                   ['romance', 'family', 'fantasy']               ['US']   
3        ['animation', 'drama', 'family', 'fantasy']               ['US']   
4                    ['family', 'fantasy', 'action']         ['GB', 'US']   

   seasons    imdb_id  imdb_score  imdb_votes  tmdb_popularity  tmdb_score  \
0      NaN  tt0041094         6.9     16497.0           20.116       6.500   
1      NaN  tt0042332         7.3    164222.0           93.547       7.035   
2      NaN  tt0128996         6.7      8944.0            9.493       6.400   
3      NaN  tt0033563         7.2    134997.0           54.265       6.999   
4      NaN  tt3861390         6.3     73010.0           23.271       6.661   

                                          director  \
0         Jack Kinney, Clyde Geronimi, James Algar   
1  Wilfred Jackson, Hamilton Luske, Clyde Geronimi   
2  Wilfred Jackson, Hamilton Luske, Clyde Geronimi   
3                                       Tim Burton   
4                                       Tim Burton   

                                            country               from rating  \
0                                     United States  November 12, 2019      G   
1                                     United States  November 12, 2019      G   
2                                     United States  November 12, 2019      G   
3  United States, United Kingdom, Australia, Canada  November 12, 2019     PG   
4  United States, United Kingdom, Australia, Canada  November 12, 2019     PG   

  duration  
0   69 min  
1   77 min  
2   77 min  
3  117 min  
4  117 min

In [407]:
disney_titles.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1070 entries, 0 to 1069
Data columns (total 20 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            1070 non-null   int64  
 1   id                    1070 non-null   object 
 2   title                 1070 non-null   object 
 3   show_type             1070 non-null   object 
 4   release_year          1070 non-null   int64  
 5   age_certification     854 non-null    object 
 6   runtime               1070 non-null   int64  
 7   genres                1070 non-null   object 
 8   production_countries  1070 non-null   object 
 9   seasons               263 non-null    float64
 10  imdb_id               807 non-null    object 
 11  imdb_score            792 non-null    float64
 12  imdb_votes            791 non-null    float64
 13  tmdb_popularity       1067 non-null   float64
 14  tmdb_score            1027 non-null   float64
 15  director             

Come abbiamo visto, i due dataset hanno molte colonne in comune, ma anche alcune differenze.

Proviamo comunque ad integrare le due tabelle, creandone una terza che metta insieme i dati della prima e quelli della seconda. 

**tip**
Dovete "concatenare" i due dataFrame di pandas. 

Provate a cercare la funzione corretta e scrivete il codice necessario. 

In [408]:
# ... completare qui ...
integrated_titles = pd.concat([netflix_titles, disney_titles])

Adesso, stampiamo un po' di informazioni sul dataset integrato (come ad esempio le dimensioni) e diamo un'occhiata allo schema.

In [409]:
# ... completare qui ...

# Stampa lo schema
# integrated_titles.info();

# Stampa le dimensioni
print(netflix_titles.shape);
print(disney_titles.shape);
print(integrated_titles.shape);

(3645, 23)
(1070, 20)
(4715, 25)


Nonostante le differenze tra le due tabelle, la funzione concat concatena le due tabelle, mettendo a Null i valori delle colonne presenti solo in una delle due ed in corrispondenza delle righe provenienti dal dataset in cui la colonna non è presente. Proviamo a verificare la presenza di Null

In [410]:
# ... completare qui ...

# integrated_titles.isnull()

# Per avere un riepilogo "in numeri"
integrated_titles.isnull().sum()

Unnamed: 0                 0
id                         0
title                      0
type                    1070
release_year               0
age_certification       1858
runtime                    0
genres                     0
production_countries       0
seasons                 2999
imdb_id                  491
imdb_score               545
imdb_votes               550
tmdb_popularity           36
tmdb_score               159
director                1695
cast                    1435
country                  505
date_added              1070
rating                     2
duration                   1
listed_in               1070
description             1070
show_type               3645
from                    3646
dtype: int64

Ad esempio la colonna **listed_in**, che è presente solo in netflix_titles, ha 1070 valori Null, che corrisponde al numero di righe della tabella disney_titles. 

Decidiamo di eliminare eventuali colonne non comuni. Possiamo farlo **prima** dell'integrazione, oppure direttamente sulla tabella concatenata. 

**Tip**: confrontate i due schemi per capire quali siano le colonne non comuni. Attenzione!!! Potrebbero anche esserci colonne che sembrano non comuni, ma in realtà contengono la stessa tipologia di informazione e hanno solo nomi diversi!!

In [411]:
# ... completare qui ...
print("Prima:")
print(integrated_titles.shape)
integrated_titles = integrated_titles.drop(columns=["listed_in", "director", "description", "cast"], axis=1)

# Stampo le dimensioni della nuova tabella (ci si aspetta 4 colonne in meno)
print("Dopo:")
print(integrated_titles.shape)

Prima:
(4715, 25)
Dopo:
(4715, 21)


Andiamo ora alla ricerca di eventuali colonne che contengono la stessa informazione. Guardate in particolare (type, show_type),  (country, production_countries), (date_added, from)

In [412]:
# ...completare qui ...
index = integrated_titles["show_type"].notnull()
values = integrated_titles["show_type"][index]

### loc permette di sovrascrivere senza warning, sovrascrivo la colonna type
# (type, show_type)
integrated_titles.loc[index, "type"] = values
integrated_titles = integrated_titles.drop(columns=["show_type"], axis=1)
integrated_titles["type"].isnull().sum()

### Ripeto per le altre due coppie
# (country, production_countries)
index = integrated_titles["production_countries"].notnull()
values = integrated_titles["production_countries"][index]
integrated_titles.loc[index, "country"] = values
integrated_titles = integrated_titles.drop(columns=["production_countries"], axis=1)
integrated_titles["country"].isnull().sum()

# (date_added, from)
index = integrated_titles["from"].notnull()
values = integrated_titles["from"][index]
integrated_titles.loc[index, "date_added"] = values
integrated_titles = integrated_titles.drop(columns=["from"], axis=1)
integrated_titles["date_added"].isnull().sum()

1

Infine, ci sono colonne che già ora possiamo giudicare inutili?

per buttare via una colonna, usiamo la funzione drop di pandas 

In [413]:
# ...completare qui ...
#integrated_titles.info()

print("Prima:")
print(integrated_titles.shape)
# Oltre quelle precedentemente rimosse, rimuovo 'age_certification' non utile ai fini di questo notebook
integrated_titles = integrated_titles.drop(columns=["age_certification"], axis=1)
print("Dopo:")
print(integrated_titles.shape)

Prima:
(4715, 18)
Dopo:
(4715, 17)


### Carichiamo altre due tabelle, che forniscono fonti di informazioni aggiuntive

In [414]:
netflix_credits = pd.read_csv("netflix_credits.csv")
disney_credits = pd.read_csv("disney_credits.csv")

come sempre, stampiamo le prime righe di ciascun dataFrame 

In [415]:
netflix_credits.head()

person_id       id                name                  character   role
0       3180  tm82169  Sylvester Stallone      Robert "Rocky" Balboa  ACTOR
1       7748  tm82169         Talia Shire  Adrianna "Adrian" Pennino  ACTOR
2       7749  tm82169          Burt Young             Paulie Pennino  ACTOR
3       7750  tm82169       Carl Weathers               Apollo Creed  ACTOR
4       7751  tm82169    Burgess Meredith            Mickey Goldmill  ACTOR

In [416]:
disney_credits.head()

person_id       id            name         character   role
0      35549  tm89464  Maureen O'Hara      Doris Walker  ACTOR
1      57832  tm89464      John Payne       Fred Gailey  ACTOR
2      57833  tm89464    Edmund Gwenn      Kris Kringle  ACTOR
3      25096  tm89464    Natalie Wood      Susan Walker  ACTOR
4      27185  tm89464     Porter Hall  Granville Sawyer  ACTOR

Possiamo osservare che le tabelle credits e titles, hanno informazioni complementari.  In particolare, credits riporta delle informazinoi aggiuntive riguardanti attori e carattere interpretato. 

Come prima, procediamo con l'integrazione delle due tabelle, che questa volta è più facile (lo schema è proprio lo stesso!)

In [417]:
#completare qui

# Concateno (aggiungo righe) da una tabella all'altra
integrated_credits = pd.concat([netflix_credits, disney_credits])

# Le colonne erano e rimangono le medesime, sommo le righe
# integrated_credits.info()
print(netflix_credits.shape)
print(disney_credits.shape)
print(integrated_credits.shape)

(80383, 5)
(30408, 5)
(110791, 5)


Data la complementarietà delle informazioni che troviamo nelle cartelle titles e credits, possiamo decidere di fare un **join** tra le due tabelle, per includere tutte le informazioni in un unico dataset. 


In [418]:
# ...completare qui usando la funzione merge per fare il join
# In questo caso il merge unisce le tabelle solo dove presente l'attributo di join, dunque il numero di righe risultante può variare
final_table = pd.merge(integrated_titles, integrated_credits, how='inner', left_on='id', right_on='id')

verifichiamo cosa sia successo guardando le dimensioni di ciascun dataframe

In [419]:
#completare qui 
# Notiamo che le righe risultanti sono diverse dal numero ottenuto dalla concatenazione
print(final_table.shape)
final_table.info()

(66189, 21)
<class 'pandas.core.frame.DataFrame'>
Int64Index: 66189 entries, 0 to 66188
Data columns (total 21 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Unnamed: 0       66189 non-null  int64  
 1   id               66189 non-null  object 
 2   title            66189 non-null  object 
 3   type             66189 non-null  object 
 4   release_year     66189 non-null  int64  
 5   runtime          66189 non-null  int64  
 6   genres           66189 non-null  object 
 7   seasons          11529 non-null  float64
 8   imdb_id          62221 non-null  object 
 9   imdb_score       61938 non-null  float64
 10  imdb_votes       61901 non-null  float64
 11  tmdb_popularity  66189 non-null  float64
 12  tmdb_score       65664 non-null  float64
 13  country          66189 non-null  object 
 14  date_added       66188 non-null  object 
 15  rating           66159 non-null  object 
 16  duration         66187 non-null  object 
 17  

vi tornano le dimensioni?

In [420]:
final_table.head()

Unnamed: 0       id  title   type  release_year  runtime  \
0           1  tm82169  Rocky  MOVIE          1976      119   
1           1  tm82169  Rocky  MOVIE          1976      119   
2           1  tm82169  Rocky  MOVIE          1976      119   
3           1  tm82169  Rocky  MOVIE          1976      119   
4           1  tm82169  Rocky  MOVIE          1976      119   

               genres  seasons    imdb_id  imdb_score  ...  tmdb_popularity  \
0  ['drama', 'sport']      NaN  tt0075148         8.1  ...           99.713   
1  ['drama', 'sport']      NaN  tt0075148         8.1  ...           99.713   
2  ['drama', 'sport']      NaN  tt0075148         8.1  ...           99.713   
3  ['drama', 'sport']      NaN  tt0075148         8.1  ...           99.713   
4  ['drama', 'sport']      NaN  tt0075148         8.1  ...           99.713   

   tmdb_score  country      date_added rating duration person_id  \
0       7.783   ['US']  August 1, 2019     PG  120 min      3180   
1       7.783   ['US']  August 1, 2019     PG  120 min      7748   
2       7.783   ['US']  August 1, 2019     PG  120 min      7749   
3       7.783   ['US']  August 1, 2019     PG  120 min      7750   
4       7.783   ['US']  August 1, 2019     PG  120 min      7751   

                 name                  character   role  
0  Sylvester Stallone      Robert "Rocky" Balboa  ACTOR  
1         Talia Shire  Adrianna "Adrian" Pennino  ACTOR  
2          Burt Young             Paulie Pennino  ACTOR  
3       Carl Weathers               Apollo Creed  ACTOR  
4    Burgess Meredith            Mickey Goldmill  ACTOR  

[5 rows x 21 columns]

In [421]:
print(final_table.columns)

Index(['Unnamed: 0', 'id', 'title', 'type', 'release_year', 'runtime',
       'genres', 'seasons', 'imdb_id', 'imdb_score', 'imdb_votes',
       'tmdb_popularity', 'tmdb_score', 'country', 'date_added', 'rating',
       'duration', 'person_id', 'name', 'character', 'role'],
      dtype='object')


e visualizziamo gli eventuali elementi mancanti

In [422]:
final_table.isnull().sum()

Unnamed: 0             0
id                     0
title                  0
type                   0
release_year           0
runtime                0
genres                 0
seasons            54660
imdb_id             3968
imdb_score          4251
imdb_votes          4288
tmdb_popularity        0
tmdb_score           525
country                0
date_added             1
rating                30
duration               2
person_id              0
name                   0
character           7059
role                   0
dtype: int64

concentriamoci sulla colonna date_added, intanto visualizzandola:

In [423]:
#completare qui 
print(final_table['date_added'])

0            August 1, 2019
1            August 1, 2019
2            August 1, 2019
3            August 1, 2019
4            August 1, 2019
                ...        
66184      October 20, 2021
66185    September 15, 2021
66186          May 28, 2021
66187          May 28, 2021
66188      October 29, 2021
Name: date_added, Length: 66189, dtype: object


vediamo che giorni mesi e anni sono presenti. Potrebbe essere utile, però, separare queste informazioni, incorporandole in tre colonne diverse. 

Facciamolo insieme, con due diversi passi:

1. dobbiamo trasformare questa colonna in formato 'datetime' utilizzando la funzione di pandas 'to_datetime'

In [424]:
#completare qui 
final_table['date_added'] = pd.to_datetime(final_table['date_added'])

In [425]:
print(final_table['date_added'])

0       2019-08-01
1       2019-08-01
2       2019-08-01
3       2019-08-01
4       2019-08-01
           ...    
66184   2021-10-20
66185   2021-09-15
66186   2021-05-28
66187   2021-05-28
66188   2021-10-29
Name: date_added, Length: 66189, dtype: datetime64[ns]


poi, possiamo separare le varie entrate in tre nuove colonne, che si chiameranno day, month e year. Bisogna fare colonna.dt.year e cambiare poi l'ultimo metodo 

In [426]:
#completare qui 
final_table['day'] = final_table['date_added'].dt.day
final_table['month'] = final_table['date_added'].dt.month
final_table['year'] = final_table['date_added'].dt.year

final_table.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 66189 entries, 0 to 66188
Data columns (total 24 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   Unnamed: 0       66189 non-null  int64         
 1   id               66189 non-null  object        
 2   title            66189 non-null  object        
 3   type             66189 non-null  object        
 4   release_year     66189 non-null  int64         
 5   runtime          66189 non-null  int64         
 6   genres           66189 non-null  object        
 7   seasons          11529 non-null  float64       
 8   imdb_id          62221 non-null  object        
 9   imdb_score       61938 non-null  float64       
 10  imdb_votes       61901 non-null  float64       
 11  tmdb_popularity  66189 non-null  float64       
 12  tmdb_score       65664 non-null  float64       
 13  country          66189 non-null  object        
 14  date_added       66188 non-null  datet

ora visualizziamo il risultato

In [427]:
final_table.head()

Unnamed: 0       id  title   type  release_year  runtime  \
0           1  tm82169  Rocky  MOVIE          1976      119   
1           1  tm82169  Rocky  MOVIE          1976      119   
2           1  tm82169  Rocky  MOVIE          1976      119   
3           1  tm82169  Rocky  MOVIE          1976      119   
4           1  tm82169  Rocky  MOVIE          1976      119   

               genres  seasons    imdb_id  imdb_score  ...  date_added  \
0  ['drama', 'sport']      NaN  tt0075148         8.1  ...  2019-08-01   
1  ['drama', 'sport']      NaN  tt0075148         8.1  ...  2019-08-01   
2  ['drama', 'sport']      NaN  tt0075148         8.1  ...  2019-08-01   
3  ['drama', 'sport']      NaN  tt0075148         8.1  ...  2019-08-01   
4  ['drama', 'sport']      NaN  tt0075148         8.1  ...  2019-08-01   

   rating  duration person_id                name                  character  \
0      PG   120 min      3180  Sylvester Stallone      Robert "Rocky" Balboa   
1      PG   120 min      7748         Talia Shire  Adrianna "Adrian" Pennino   
2      PG   120 min      7749          Burt Young             Paulie Pennino   
3      PG   120 min      7750       Carl Weathers               Apollo Creed   
4      PG   120 min      7751    Burgess Meredith            Mickey Goldmill   

    role  day month    year  
0  ACTOR  1.0   8.0  2019.0  
1  ACTOR  1.0   8.0  2019.0  
2  ACTOR  1.0   8.0  2019.0  
3  ACTOR  1.0   8.0  2019.0  
4  ACTOR  1.0   8.0  2019.0  

[5 rows x 24 columns]

ora creiamo una copia della tabella, contenente soltanto i film.


In [428]:
#completare qui 
index = (final_table['type'] == 'MOVIE') & (final_table['duration'].str.contains('min'))
movies = final_table[index].copy()

print(movies.shape)

# Check aggiuntivo se ogni riga di movies ha season NaN
print(movies['seasons'].notnull().values.sum()) # mi aspetto zero valori non nulli

(54252, 24)
0


a questo punto, possiamo modificare la durata in ore, per prima cosa eliminiamo la parola minuti 

In [429]:
#completare qui 
movies['duration'] = movies['duration'].replace(' min', '', regex=True)
#print(movies['duration'].to_string(index=False))

quindi dividere per 60 (dopo conversione float)

In [430]:
#completare qui 
movies['duration'] = movies['duration'].astype(float)/60.0

e finalmente, aggiungere un suffisso per indicare le ore, per esempio 'hours', tornando a stringa

In [431]:
#completare qui 
# Ottengo il suffisso con una sempice concatenazione fra stringhe
movies['duration'] = movies['duration'].to_string(index=False) + " hours"

stampiamo il risultato finale 

In [432]:
print(movies['duration'])

0        2.000000\n2.000000\n2.000000\n2.000000\n2.0000...
1        2.000000\n2.000000\n2.000000\n2.000000\n2.0000...
2        2.000000\n2.000000\n2.000000\n2.000000\n2.0000...
3        2.000000\n2.000000\n2.000000\n2.000000\n2.0000...
4        2.000000\n2.000000\n2.000000\n2.000000\n2.0000...
                               ...                        
66178    2.000000\n2.000000\n2.000000\n2.000000\n2.0000...
66179    2.000000\n2.000000\n2.000000\n2.000000\n2.0000...
66180    2.000000\n2.000000\n2.000000\n2.000000\n2.0000...
66181    2.000000\n2.000000\n2.000000\n2.000000\n2.0000...
66188    2.000000\n2.000000\n2.000000\n2.000000\n2.0000...
Name: duration, Length: 54252, dtype: object


concentriamoci sulla colonna release_year. Innanzitutto eliminiamo le righe dove questa informazinoe non sia presente. Usiamo la funzione dropna di pandas 
dropna(axis = , subset = [nome colonna])

In [433]:
#completare qui 
print(movies.isnull().sum())
final_table_clean = movies.dropna(axis=0, subset=['release_year'])

Unnamed: 0             0
id                     0
title                  0
type                   0
release_year           0
runtime                0
genres                 0
seasons            54252
imdb_id             3339
imdb_score          3541
imdb_votes          3578
tmdb_popularity        0
tmdb_score           392
country                0
date_added             0
rating                 3
duration               0
person_id              0
name                   0
character           5620
role                   0
day                    0
month                  0
year                   0
dtype: int64


In [434]:
final_table_clean.isnull().sum()

Unnamed: 0             0
id                     0
title                  0
type                   0
release_year           0
runtime                0
genres                 0
seasons            54252
imdb_id             3339
imdb_score          3541
imdb_votes          3578
tmdb_popularity        0
tmdb_score           392
country                0
date_added             0
rating                 3
duration               0
person_id              0
name                   0
character           5620
role                   0
day                    0
month                  0
year                   0
dtype: int64

vediamo ora la colonna release_year_y

In [435]:
# print(final_table_clean['release_year_y'])

# La colonna corretta ha qui nome 'release_year'
print(final_table_clean['release_year'])

# Nota: nella tabella movies nessuna riga presentava un valore mancante per 'release_year', dunque l'operazione di drop di righe con valori mancanti non fa nulla 

0        1976
1        1976
2        1976
3        1976
4        1976
         ... 
66178    2021
66179    2021
66180    2021
66181    2021
66188    2022
Name: release_year, Length: 54252, dtype: int64
